Sources/libreprl/libreprl-posix.c

```diff
@@ -410,11 +411,15 @@ int reprl_execute(struct reprl_context* ctx, const char* script, uint64_t script
     }
     
     // Poll succeeded, so there must be something to read now (either the status or EOF).
+    struct {
```

Comment:
So since these changes to the REPRL interface are not backwards compatible, they will break all the existing REPRL implementations for the other engines. I don't think we need to fix them, but it'd be nice if we could at least print a message along the lines of "The target engine uses an old version of the REPRL protocol, please update it to the latest version".

My suggestion would be the following:

1. Add this code at the start of this file (and I guess into the client-side code)
```C
// The REPRL protocol version. Must be identical between server and client.
// Bump this up whenever there are non-backwards-compatible changes to the protocol
#define REPRL_PROTOCOL_VERSION 2
```

2. Change this code: https://github.com/googleprojectzero/fuzzilli/blob/3af52d7f2a58754097b2920b9c9f5ab30dc63d37/Sources/libreprl/libreprl-posix.c#L262 to something like
```C
    char handshake[5] = { 0 };
    if (read(ctx->ctrl_in, handshake, 4) != 4) {
        reprl_terminate_child(ctx);
        return reprl_error(ctx, "Did not receive handshake message from child: %s", strerror(errno));
    }
    
    if (strncmp(handshake, "HELO", 4) == 0) {
        // Client is using protocol version 1.
        reprl_terminate_child(ctx);
        return reprl_error(ctx, "The client uses an old version of the REPRL protocol, please update it to the latest version");
    }

    if (strncmp(handshake, "RPRL", 4) != 0) {
        reprl_terminate_child(ctx);
        return reprl_error(ctx, "Received invalid handshake message from child: %s", helo);
    }

    int client_protocol_version;
    static_assert(sizeof(client_protocol_version) == 4);
    if (read(ctx->ctrl_in, &client_protocol_version, 4) != 4) {
        reprl_terminate_child(ctx);
        return reprl_error(ctx, "Did not receive client protocol version from child: %s", strerror(errno));
    }

    if (client_protocol_version != REPRL_PROTOCOL_VERSION) {
        reprl_terminate_child(ctx);
        return reprl_error(ctx, "The client uses an old version of the REPRL protocol, please update it to the latest version");
    }

    // The server side just sends back the handshake message, but not the protocol version.
    if (write(ctx->ctrl_out, handshake, 4) != 4) {
        reprl_terminate_child(ctx);
        return reprl_error(ctx, "Failed to send handshake reply message to child: %s", strerror(errno));
    }
```

3. Update the client-side code accordingly, i.e. send `"RPRL"` (or some other string) instead of `"HELO"` and send the protocol version number afterwards.

WDYT about this?